modules/renter/uploadheap.go

```diff
@@ -235,14 +237,44 @@ func (r *Renter) buildUnfinishedChunks(f *file, hosts map[string]struct{}) []*un
 func (r *Renter) managedBuildChunkHeap(hosts map[string]struct{}) {
 	// Loop through the whole set of files and get a list of chunks to add to
 	// the heap.
-	id := r.mu.Lock()
+	id := r.mu.RLock()
+	goodForRenew := make(map[types.FileContractID]bool)
+	offline := make(map[types.FileContractID]bool)
 	for _, file := range r.files {
+		file.mu.RLock()
+		// Build maps that map contract id to its offline and goodForRenew
+		// status
+		contractIDs := make(map[types.FileContractID]struct{})
+		for cid := range file.contracts {
+			contractIDs[cid] = struct{}{}
+		}
+		for cid := range contractIDs {
```

Comment:
Is the `contractIDs` map still necessary? Can't you just iterate through `file.contracts`?